pkg/virt-launcher/virtwrap/live-migration-source.go

```diff
@@ -502,17 +507,37 @@ func (m *migrationMonitor) processInflightMigration(dom cli.VirDomain, stats *li
 		// If we were to abort the migration due to a timeout while in post copy,
 		// then it would result in that active state being lost.
 
-	case m.shouldTriggerPostCopy(elapsed):
-		logger.Info("Starting post copy mode for migration")
-		// if a migration has stalled too long, post copy will be
-		// triggered when allowPostCopy is enabled
-		err := dom.MigrateStartPostCopy(uint32(0))
-		if err != nil {
-			logger.Reason(err).Error("failed to start post migration")
-			return nil
-		}
+	case m.shouldForceMigrationToComplete(elapsed) && !m.isPausedMigration():
 
-		m.l.updateVMIMigrationMode(v1.MigrationPostCopy)
+		if m.options.AllowPostCopy {
+			logger.Info("Starting post copy mode for migration")
+			// if a migration has stalled too long, post copy will be
+			// triggered when allowPostCopy is enabled
+			err := dom.MigrateStartPostCopy(uint32(0))
+			if err != nil {
+				logger.Reason(err).Error("failed to start post migration")
+				return nil
+			}
+
+			m.l.updateVMIMigrationMode(v1.MigrationPostCopy)
+		} else {
```

Comment:
I think my suggestion about using a strategy rather than a bool can simplify this code significantly